tests/test_inprocess.py

```diff
@@ -0,0 +1,160 @@
+# SOFTWARE.
+
+import os
+import subprocess
+from typing import TYPE_CHECKING
+
+import pytest
+
+if TYPE_CHECKING:
+    from ansys.mapdl.core import Mapdl
+
+from conftest import ON_LOCAL
+
+
+class MapdlInProcessRunner:
+    def __init__(self, wdir: str, exec_path: str | None = None) -> None:
+        self.wdir = str(wdir)
+        self.completed_process: subprocess.CompletedProcess[bytes] | None = None
+        self._exec_path: str | None = exec_path
+
+    @property
+    def exec_path(self) -> str:
+        if self._exec_path is None:
+
+            self._exec_path = os.getenv("PYMAPDL_MAPDL_EXEC")
+            if self._exec_path is None:
+                from ansys.tools.path import get_mapdl_path
+
+                version: float | None = (
+                    float(os.getenv("PYMAPDL_MAPDL_VERSION", 0)) or None
+                )
+                self._exec_path = get_mapdl_path(version=version)
+
+        if self._exec_path is None:
+            raise ValueError(
+                "MAPDL executable path is not set. Set the PYMAPDL_MAPDL_EXEC "
+                "environment variable or use get_mapdl_path() to find it."
+            )
+
+        return self._exec_path
+
+    @property
+    def stdout(self) -> str | None:
+        if self.completed_process:
+            return self.completed_process.stdout.decode()
+
+    @property
+    def stderr(self) -> str | None:
+        if self.completed_process:
+            return self.completed_process.stderr.decode()
+
+    def run(self, cmds: str) -> str:
+        """Run commands in MAPDL on batch mode."""
+        with open(os.path.join(self.wdir, "input.mac"), "w") as f:
+            f.write(cmds)
+
+        try:
+            args = [
+                self.exec_path,
+                "-b",
+                "-i",
+                "input.mac",
+                "-o",
+                "out.out",
+                os.getenv("PYMAPDL_ADDITIONAL_SWITCHES", ""),
+            ]
+
+            self.completed_process = subprocess.run(
+                args=args,
+                cwd=self.wdir,
+                check=True,
+                capture_output=True,
+                # it does not support shell=True, because it does not
+                # generate the out.out file
+                shell=False,
+            )
+
+        except subprocess.CalledProcessError as e:
+            error_msg = (
+                f"MAPDL execution failed with return code {e.returncode}.\n"
+                f"Command: {' '.join(e.cmd)}\n"
+                f"Stdout:\n{e.stdout.decode() if e.stdout else ''}\n"
+                f"Stderr:\n{e.stderr.decode() if e.stderr else ''}\n"
+            )
+            pytest.fail(error_msg)
+
+        with open(os.path.join(self.wdir, "out.out"), "r") as f:
+            self.output = f.read()
+
+        return self.output
+
+
+@pytest.fixture()
+def mapdl_inprocess(mapdl: "Mapdl", tmp_path) -> MapdlInProcessRunner:
```

Comment:
make a `mapdl_version` fixture which in the first place can be based on the mapdl client instance, but in the future can be refactored so that when running a test that doesn't use that fixture, it can find out the version through more direct means.

Reply:
Done.